cranelift/interpreter/src/interpreter.rs

```diff
@@ -240,6 +240,29 @@ impl<'a> InterpreterState<'a> {
             _ => &self.frame_stack[num_frames - 1],
         }
     }
+
+    fn is_address_within_stack_slot_range(&self, addr: &Address, ty: &Type) -> bool {
+        let size = ty.bytes() as u64;
+        let addr_start = addr.offset;
+        let addr_end = addr_start + size;
+        let stack_slots = &self.get_current_function().sized_stack_slots;
+        let mut slot_offset = self.frame_offset as u64;
+        let ss = stack_slots
+            .values()
+            .find(|ss| {
+                let next_slot_offset = slot_offset + ss.size as u64;
+                let addr_start_is_in_ss_range =
+                    (slot_offset..next_slot_offset).contains(&addr_start);
+                if addr_start_is_in_ss_range {
+                    true
+                } else {
+                    slot_offset = next_slot_offset;
+                    false
+                }
+            })
+            .expect("address to fall in the range of atleast one stack slot");
```

Comment:
We probably don't want to panic here. We can reach this condition by doing something like:

```
v0 = stack_addr ss0
v1 = iadd_imm v0, 0xFFFFFF
```

Which would normally generate a OutOfBoundsLoad / Store instead of panicking

Reply:
I originally thought that since the check that precedes this is the check for storing/loading beyond the stack, it would be caught. However, in hindsight it does not appear to be sufficient. I will gladly change it.